File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/server/ChannelJoinSuccessPacket.java
package pe.nanamochi.banchus.packets.server;

import java.io.IOException;
import java.io.OutputStream;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.core.ServerPacket;
import pe.nanamochi.banchus.io.IDataWriter;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ChannelJoinSuccessPacket implements ServerPacket {
  private String name;

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_CHANNEL_JOIN_SUCCESS;
  }

  @Override
  public void write(IDataWriter writer, OutputStream stream) throws IOException {
    writer.writeString(stream, name);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/ChannelJoinSuccessPacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.core.ServerPacket
import pe.nanamochi.banchus.io.DataWriter

class ChannelJoinSuccessPacket(var name: String = "") : ServerPacket() {
    override val type = PacketType.BANCHO_CHANNEL_JOIN_SUCCESS

    override fun write(writer: DataWriter, out: OutputStream) {
        writer.writeString(out, name)
    }
}
